piDIOComm.c

```diff
@@ -223,54 +222,34 @@ INT32U piDIOComm_sendCyclicTelegram(INT8U i8uDevice_p)
 			    sRequest_l.ai8uData[0], sRequest_l.ai8uData[1]);
 	}
 #endif
-	memcpy(last_out[i8uAddress], data_out, sizeof(data_out));
-
-	ret = piIoComm_send((INT8U *) & sRequest_l, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
-	if (ret == 0) {
-		len_l = 3 * sizeof(INT16U) + i8uNumCounter[i8uAddress] * sizeof(INT32U);
-
-		ret = piIoComm_recv((INT8U *) & sResponse_l, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
-		if (ret > 0) {
-			if (piIoComm_response_valid(&sResponse_l, i8uAddress, len_l)) {
-				memcpy(&data_in[0], sResponse_l.ai8uData, 3 * sizeof(INT16U));
-				memset(&data_in[6], 0, 64);
-				p = 0;
-				for (i = 0; i < 16; i++) {
-					if (i16uCounterAct[i8uAddress] & (1 << i)) {
-						memcpy(&data_in[3 * sizeof(INT16U) + i * sizeof(INT32U)],
-						       &sResponse_l.ai8uData[3 * sizeof(INT16U) + p * sizeof(INT32U)],
-						       sizeof(INT32U));
-						p++;
-					}
-				}
+	memcpy(last_out[i8uAddress], out_buf, sizeof(out_buf));
 
-				rt_mutex_lock(&piDev_g.lockPI);
-				memcpy(piDev_g.ai8uPI + RevPiDevice_getDev(i8uDevice_p)->i16uInputOffset, data_in,
-				       sizeof(data_in));
-				rt_mutex_unlock(&piDev_g.lockPI);
+	rcv_len = 3 * sizeof(INT16U) + i8uNumCounter[i8uAddress] * sizeof(INT32U);
 
-#ifdef DEBUG_DEVICE_DIO
-				if (last_in[i8uAddress][0] != sResponse_l.ai8uData[0]
-				    || last_in[i8uAddress][1] != sResponse_l.ai8uData[1]) {
-					last_in[i8uAddress][0] = sResponse_l.ai8uData[0];
-					last_in[i8uAddress][1] = sResponse_l.ai8uData[1];
-					pr_info_dio("dev %2d: recv cyclic Data addr %d input 0x%02x 0x%02x\n\n",
-						    i8uAddress, RevPiDevice_getDev(i8uDevice_p].i16uInputOffset,
-						    sResponse_l.ai8uData[0], sResponse_l.ai8uData[1]);
-				}
-#endif
-			} else {
-				i32uRv_l = 1;
-			}
-		} else {
-			i32uRv_l = 2;
-			pr_info_dio("dev %2d: recv ioprotocol timeout error exp %d\n",
-				    i8uAddress, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
+	ret = pibridge_req_io(i8uAddress, cmd, snd_buf, snd_len, in_buf,
+			      rcv_len);
+	if (ret) {
+		pr_info_dio("dev %2d: send ioprotocol send error %d\n", i8uAddress, ret);
```

Comment:
The message is a little misleading, this could also be a receive error now, not just a send error. Also the message should use `KERN_WARN` severity and be ratelimited.